### ISSUE TITLE
Allow custom hooks to return stable results

### ISSUE THREAD

**Issue by FezVrasta** [facebook/react#22539]
Right now only `useRef` and other first-party hooks are allowed to return a stable result that is not required to be listed in the dependencies array of other hooks.

We have the `react-hooks/exhaustive-deps`' `additionalHooks` option to specify additional hooks that take dependencies arrays, but we don't have a way to describe a hook that returns a stable result.

An example of such hook could be:

```js
const useLatest = (value) => {
  const ref = useRef(value);
  ref.current = value;
  return ref;
};
```

It would be extremely useful being able to define custom hooks like this with a rule that could look like the following:

```
"react-hooks/exhaustive-deps": [
  "warn",
  {
    "stableKnownHookValues": "(useLatest|useSomethingElse)"
  }
]
```

I'd be willing to work on a PR if this idea is deemed useful.

**Comment by mcrowder65** [comment]
@FezVrasta this is related: https://github.com/facebook/react/pull/20513

**Comment by grncdr** [comment]
So...

- I opened the same issue in September 2019 (https://github.com/facebook/react/issues/16873)
- I modified eslint-plugin-react-hooks and have been using my version ever since.
- My fork is available on NPM as `@grncdr/eslint-plugin-react-hooks` for others to use (I think there are a few users)
- The React team has never responded on that issue, to indicate whether they think it's a good idea, so I never opened a PR.
- The code in PR #20513 is more-or-less based on my fork afaict.

Anyways, if you want to try it out, it's easy, just replace `eslint-plugin-react-hooks` with `@grncdr/eslint-plugin-react-hooks` and configure the `staticHooks` property as shown in this comment: https://github.com/facebook/react/issues/16873#issuecomment-536346885
